pkg/artifacts/fetch.go

```diff
@@ -197,7 +218,84 @@ func FetchLastNTestResults(test_matrix v1.MatrixSpec, matrix_name, prow_name str
 	return build_ids, test_results, err
 }
 
-func FetchTestStepResult(test_matrix v1.MatrixSpec, test_result v1.TestResult, filename string, filetype ArtifactType) (ArtifactResult, error) {
-	step_filenane := fmt.Sprintf("artifacts/%s/%s/%s", test_result.TestSpec.TestName, test_matrix.ProwStep, filename)
-	return fetchTestResult(test_matrix, test_result.TestSpec.ProwName, test_result.BuildId, step_filenane, filetype)
+func FetchTestStepResult(test_matrix *v1.MatrixSpec, test_spec *v1.TestSpec, build_id string, filename string, filetype ArtifactType) (ArtifactResult, error) {
+	var prow_step = test_matrix.ProwStep
+	if test_spec.ProwStep != "" {
+		// override test_matrix.ProwStep if ProwStep is test_spec.ProwStep is specified
+		prow_step = test_spec.ProwStep
+	}
+	step_filenane := fmt.Sprintf("artifacts/%s/%s/%s", test_spec.TestName, prow_step, filename)
+	return fetchTestResult(test_matrix, test_spec.ProwName, build_id, step_filenane, filetype)
+}
+
+func FetchTestToolboxSteps(test_matrix *v1.MatrixSpec, test_spec *v1.TestSpec, build_id string) ([]string, error) {
+	html_toolbox_steps, err := FetchTestStepResult(test_matrix, test_spec, build_id, "artifacts/", TypeHtml)
+	if err != nil {
+		return []string{}, err
+	}
+
+	toolbox_steps, err := listFilesInDirectory(html_toolbox_steps.Html, true, false)
+	if err != nil {
+		return []string{}, fmt.Errorf("error fetching toolbox steps: %v", err)
+	}
+
+	return toolbox_steps, nil
+}
+
+func FetchTestToolboxLogs(test_matrix *v1.MatrixSpec, test_spec *v1.TestSpec, build_id string) (map[string]JsonArray, error) {
+	toolbox_steps, err := FetchTestToolboxSteps(test_matrix, test_spec, build_id)
+	if err != nil {
+		fmt.Println(err)
+		return map[string]JsonArray{}, err
+	}
+	logs := map[string]JsonArray{}
+
+	for _, toolbox_step := range toolbox_steps {
+		ansible_log_path := "artifacts/"+ toolbox_step + "/_ansible.log.json"
+		json_toolbox_step_logs, err := FetchTestStepResult(test_matrix, test_spec, build_id, ansible_log_path, TypeJsonArray)
+		if err != nil {
+			log.Debugf("No logs for step %s: %v", toolbox_step, err)
+			continue // ignore
```

Comment:
Can we have a better comment than just `ignore`?

Reply:
```
// no `_ansible.log.json` in the current step, meaning
// that this directory wasn't generated by a
// toolbox+ansible command. Ignore.
```

makes me wonder about "non toolbox+ansible" directores, I'm thinking in particular about [ginkgo](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-psap-ci-artifacts-release-4.8-nto-e2e-master/1409647999579066368/artifacts/e2e-master/nightly/artifacts/001__nto__run_e2e_test/nto_e2e_test.log) tests ...
That can be a future work!